virttest/utils_net.py

```diff
@@ -3008,15 +3008,25 @@ def get_ip_address_by_interface(ifname):
 
 def get_host_ip_address(params):
     """
-    returns ip address of host specified in host_ip_addr parameter If provided
-    otherwise ip address on interface specified in netdst parameter is returned
+    Returns ip address of host specified in host_ip_addr parameter if provided.
+    Otherwise ip address on interface specified in netdst parameter is returned.
+    In case of "nettype == user" "netdst" parameter is left empty, then the
+    first interface of the system is used.
     :param params
     """
     host_ip = params.get('host_ip_addr', None)
-    if not host_ip:
-        host_ip = get_ip_address_by_interface(params.get('netdst'))
-        logging.warning("No IP address of host was provided, using IP address"
-                        " on %s interface", str(params.get('netdst')))
+    if host_ip:
+        logging.debug("Use IP address at config %s=%s", 'host_ip_addr', host_ip)
+        return host_ip
+    net_dev = params.get('netdst')
+    if not net_dev:
+        net_dev = get_host_iface()
```

Comment:
@will-Do 

This won't work: net_dev = params.get('netdst', get_host_iface())

In cartesian configs : 
netdst =
params.get('netdst', smth_default) always will return empty string. It will not take "smth_default". As a key netdst is defined. It is just empty string.